crates/log-server/src/rocksdb_logstore/keys.rs

```diff
@@ -103,41 +112,39 @@ impl DataRecordKey {
         self.offset
     }
 
-    pub fn exclusive_upper_bound(loglet_id: LogletId) -> BytesMut {
-        let mut buf = BytesMut::with_capacity(Self::size());
-        KeyPrefix::new(KeyPrefixKind::DataRecord, loglet_id).encode_exclusive_upper_bound(&mut buf);
-        buf.put_u64(0);
+    pub fn exclusive_upper_bound(loglet_id: LogletId) -> [u8; Self::size()] {
+        let mut buf = [0u8; Self::size()];
+        let mut b = &mut buf[..];
+        KeyPrefix::new(KeyPrefixKind::DataRecord, loglet_id).encode_exclusive_upper_bound(&mut b);
+        (&mut b).put_u64(0);
```

Comment:
Since the array is already initialised with 0 on allocation. in line 116, this call is not necessary imho

Reply:
it's true, but it'd be an implicit assumption that future eyes would miss.